pkg/proctree/process.go

```diff
@@ -0,0 +1,383 @@
+package proctree
+
+import (
+	"sync"
+	"time"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/aquasecurity/tracee/pkg/utils/types"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// processNode is a node in the process tree representing a process.
+// Its purpose is to keep the information of the process, and the connections to other processes
+// and threads nodes.
+// The current implementation is using mutex, so all internal members are assumed to be protected
+// by it (so are not necessarily thread-safe types)
```

Comment:
You want to say here you're using mutex as a big lock for all fields within this type.